test/apps/iodemo/io_demo.cc

```diff
@@ -1059,7 +1080,12 @@ class DemoServer : public P2pDemoCommon {
         IoWriteResponseCallback *w = _callback_pool.get();
         ConnectionStat &conn_stat  = _conn_stat_map.find(conn)->second;
 
-        iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
+        if (!ucx_am_is_rndv(data_desc)) {
+            iov->init(msg->data_size, _zcopy_chunks_pool, msg->sn, opts().validate);
```

Comment:
this is hacky; instead, need to add new init function to iov class, to accept and external buffer pointer (instead of memory pool to allocate from). this buffer pointer will be ucx_am_get_data(data_desc)
same comment for dispatch_am_message() on the client

Reply:
iov takes Buffer objects which are expected to be produced from a BufferMemoryPool object. Member functions including validate()/release() are dealing with Buffer objects. Do we really want to let iov take void pointer?

Reply:
yes, and it will know that this pointer should not be released in desctructor

Reply:
done